File: algorithm/baekjoon/src/main/java/com/keunwon/algorithm/bruteforce/Problem15651.java
package com.keunwon.algorithm.bruteforce;

import java.io.BufferedWriter;
import java.io.IOException;
import java.io.OutputStreamWriter;
import java.util.Scanner;

public class Problem15651 {
    public static final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static int[] arr;

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        final int n = sc.nextInt();
        final int m = sc.nextInt();

        arr = new int[m];
        dfs(n, m, 0);

        bw.flush();
        bw.close();
    }

    public static void dfs(int n , int m, int depth) throws IOException {
        if (m == depth) {
            for (int i : arr) {
                bw.write(i + " ");
            }
            bw.write(System.lineSeparator());
            return;
        }

        for (int i = 0; i < n; i++) {
            arr[depth] = i + 1;
            dfs(n, m, depth + 1);
        }
    }
}

File: algorithm/kotlin-algorithm/src/main/kotlin/com/keunwon/algorithm/baekjoon/Problem15651.kt
package com.keunwon.algorithm.baekjoon

/**
 * Title: N과 M (3)
 * Level: 실버-3
 **/
class Problem15651 {
    private val answer = mutableListOf<String>()

    fun solution(n: Int, m: Int): List<String> {
        dfs(0, n, IntArray(m))
        return answer
    }

    private fun dfs(depth: Int, n: Int, picks: IntArray) {
        if (depth == picks.size) {
            answer.add(picks.joinToString(" "))
            return
        }

        for (i in 1..n) {
            picks[depth] = i
            dfs(depth + 1, n, picks)
        }
    }
}

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.out.bufferedWriter()

    val (n, m) = br.readLine().split(" ").map { it.toInt() }
    Problem15651().solution(n, m).forEach {
        bw.write(it)
        bw.newLine()
    }

    bw.flush()
    bw.close()
}
